openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py

```diff
@@ -2,186 +2,155 @@
+        if not prev_query_params or "offset" not in prev_query_params:
+            offset = 0
         else:
-            condition = False
-    image_store.commit()
-    logger.info(f"Total images received {image_store.total_items}")
-
-
-def _get_query_param(offset=0, default_query_param=None):
-    if default_query_param is None:
-        default_query_param = DEFAULT_QUERY_PARAMS
-    query_param = default_query_param.copy()
-    query_param.update(offset=offset)
-    return query_param
-
-
-def _get_object_json(
-    headers=None, endpoint=ENDPOINT, retries=RETRIES, query_param=None
-):
-    if headers is None:
-        headers = HEADERS.copy()
-    data = None
-    for tries in range(retries):
-        response = delay_request.get(endpoint, query_param, headers=headers)
-        try:
-            response_json = response.json()
-            if response_json and response_json.get("message", "").lower() == "success.":
-                data = response_json.get("data")
-                break
-        except (
-            RequestException,
-            AttributeError,
-            JSONDecodeError,
-            ValueError,
-            TypeError,
-        ) as e:
-            logger.error(f"Error due to {e}")
-    return data
-
-
-def _process_objects_batch(objects_batch):
-    for object_ in objects_batch:
-        rights_info = object_.get("rights_type")
-        license_url = _get_license_url(rights_info)
-        logger.debug(license_url)
-        if license_url is not None:
-            id_ = object_.get("id", "")
-            complete_object_data = _get_object_json(endpoint=ENDPOINT + str(id_))
-            if complete_object_data is None:
-                continue
-            _handle_object_data(data=complete_object_data, license_url=license_url)
-
-
-def _handle_object_data(data, license_url):
-    image_info = data.get("images")
-    if image_info is not None:
-        id_ = data.get("id", "")
-        title = data.get("title", "")
-        foreign_url = f"https://www.brooklynmuseum.org/opencollection/objects/{id_}"
-        metadata = _get_metadata(data)
-        creators = _get_creators(data)
-
-        for image in image_info:
-            foreign_id = image.get("id", "")
-            image_url = _get_image_url(image)
-            if image_url is None:
-                continue
-            height, width = _get_image_sizes(image)
-            license_info = get_license_info(license_url=license_url)
-            image_store.add_item(
-                foreign_landing_url=foreign_url,
-                image_url=image_url,
-                license_info=license_info,
-                foreign_identifier=foreign_id,
-                width=width,
-                height=height,
-                title=title,
-                meta_data=metadata,
-                creator=creators,
+            offset = prev_query_params["offset"] + self.batch_limit
+        return {**self._common_args, "offset": offset}
```

Comment:
Are we discarding the `prev_query_params` here? I think the pattern in other scripts was to use `prev_query_params` if we have them, and update them with new `offset` value (or whatever is necessary), and to create the new `query_params` with the `offset` equal to 0 if `prev_query_params`. So, in this case, there should never be the case where `offset` is not in the `prev_query_params`. 

```python
def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
    if not prev_query_params:
        return { "has_images": 1, "rights_type_permissive": 1, "limit": batch_limit, offset: 0 }
    else:
        return {**prev_query_params, offset: prev_query_params["offset"] + self.batch_limit}
````

Reply:
Ah, that's a much better approach. Thank you!